.github/workflows/ci.yml

```diff
@@ -19,23 +19,124 @@ concurrency:
   cancel-in-progress: ${{ github.event_name != 'workflow_call' }}
 
 jobs:
+  planner:
+    runs-on: ubuntu-latest
+    outputs:
+      jobs_to_run: ${{ steps.plan.outputs.jobs_to_run }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Plan jobs to run
+        id: plan
+        run: |
+          # Fetch base ref for PRs and merge_group events
+          if [ "${{ github.event_name }}" = "pull_request" ] || [ "${{ github.event_name }}" = "merge_group" ]; then
+            git fetch origin ${{ github.base_ref }} --depth=1
+            git diff --name-only origin/${{ github.base_ref }} ${{ github.sha }} > changed_files.txt
+          else
+            # For workflow_dispatch or other events, run all jobs
+
+            jobs=static-analysis,rust,kotlin,swift,website,codeql,build-artifacts,build-perf-artifacts
+            echo "jobs_to_run=$jobs" >> $GITHUB_OUTPUT
+
+            exit 0;
+          fi
+
+          echo "Changed files:"
+          cat changed_files.txt
+
+          jobs="static-analysis" # Always run static-analysis
+
+          if grep -q '^rust/' changed_files.txt; then
+            jobs="${jobs},rust,kotlin,swift,build-artifacts,build-perf-artifacts"
+          fi
+          if grep -q '^elixir/' changed_files.txt; then
+            jobs="${jobs},elixir,codeql,build-artifacts"
+          fi
+          if grep -q '^kotlin/' changed_files.txt; then
+            jobs="${jobs},kotlin"
+          fi
+          if grep -q '^swift/' changed_files.txt; then
+            jobs="${jobs},swift"
+          fi
+          if grep -q '^website/' changed_files.txt; then
+            jobs="${jobs},website,codeql"
+          fi
+
+          echo "jobs_to_run=$jobs" >> $GITHUB_OUTPUT
+
   kotlin:
+    needs: planner
+    if: contains(needs.planner.outputs.jobs_to_run, 'kotlin')
     uses: ./.github/workflows/_kotlin.yml
     secrets: inherit
   swift:
+    needs: planner
+    if: contains(needs.planner.outputs.jobs_to_run, 'swift')
     uses: ./.github/workflows/_swift.yml
     secrets: inherit
   elixir:
+    needs: planner
+    if: contains(needs.planner.outputs.jobs_to_run, 'elixir')
     uses: ./.github/workflows/_elixir.yml
   rust:
+    needs: planner
+    if: contains(needs.planner.outputs.jobs_to_run, 'rust')
     uses: ./.github/workflows/_rust.yml
     secrets: inherit
   static-analysis:
+    needs: planner
+    if: contains(needs.planner.outputs.jobs_to_run, 'static-analysis')
```

Comment:
I saw the comment above mentioning we always run this job - why the conditional here as well?